exercises/react/example/Reactor.ceylon

```diff
@@ -11,17 +13,35 @@ class Reactor() {
     assign currentValue {
       currentValue_ = currentValue;
     }
+
+    shared actual variable Cell[] dependents = [];
+    shared actual void addDependent(Cell c) {
+      dependents = dependents.withTrailing(c);
+    }
   }
 
   shared class ComputeCell satisfies Cell {
     variable Element currentValue_;
     Element() newValue;
 
+    shared actual variable Cell[] dependents = [];
+    shared actual void addDependent(Cell c) {
+      dependents = dependents.withTrailing(c);
+    }
+
     shared new single(Cell c, Element(Element) f) {
+      c.addDependent(this);
```

Comment:
this is not legal.

My alternatives are: 

* create a different function on `Reactor` that constructs a compute cell then adds it as a dependent of its dependencies. I was afraid it would be unidiomatic to expose this constructor-that-isn't-a-constructor, but perhaps it is not so bad.
* do it a different way, do not let cells know about their dependencies (why do I do it this way? only because it's the way which I did it in other language)
* force the caller to add the just-constructed `ComputeCell` as a dependent of its dependencies. I don't find this a reasonable interface.